public/locations-list-map.js

```diff
@@ -316,10 +367,186 @@ function initMap(stateFilter) {
+  }
+
+  let place = autocomplete.getPlace();
+  if (place.geometry) {
+    // Get the location object that we can map.setCenter() on.
+    let location = place.geometry.location;
+    if (location) {
+      centerMapToMarkersNearCoords(location.lat(), location.lng())
+
+    } else {
+      console.warn('Location data not found in place geometry (place.geometry.location).')
+    }
+  } else {
+    console.warn('No geometry found, skipping place');
+  }
+}
+
+
+
+/**
+ * Centers map at automatically detected coordinates using built in navigator.geolocation API.
+ */
+function centerMapToMarkersNearUser() {
+  // First check to see if the user will accept getting their location, if not, silently return
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function (position) {
+      // Use navigator provided lat/long coords to center map now.
+      centerMapToMarkersNearCoords(position.coords.latitude, position.coords.longitude);
+
+    }, function (err) {
+      // Hide the "User my location" link since we know that will not work.
+      $('#use-location').hide();
+
+    }, {
+      maximumAge: Infinity,
+      timeout: 10000
+    });
+  }
+}
+
+
+
+/**
+ * Centers map around markers nearest to an arbitrary set of latitude/longitude coordinates.
+ */
+function centerMapToMarkersNearCoords(latitude, longitude) {
+  var latlng = new google.maps.LatLng(latitude, longitude);
+
+  //Compute the distances of all markers from the user
+  var markerDistances = new Map(); // an associative array containing the marker referenced by the computed distance
+  var distances = []; // all the distances, so we can sort and then call markerDistances
+  for (const marker of markers) {
+    let distance = google.maps.geometry.spherical.computeDistanceBetween(marker.position, latlng);
+
+    // HACK: In the unlikely event that the exact same distance is computed, add one meter to the distance to give it a unique distance
+    // This could occur if a marker was added twice to the same location.
+    if (markerDistances.has(distance)) { distance = distance + 1; }
+
+    markerDistances[distance] = marker;
+
+    distances.push(distance);
+  }
+
+  // sort the distances and set bounds to closest three
+  distances.sort((a, b) => a - b);
+
+  // center the map on the user
+  const bounds = new google.maps.LatLngBounds();
+  bounds.extend(latlng);
+
+  // Extend the bounds to contain the three closest markers
+  for(let i = 0; i < 3; i++) {
+    // Get one of the closest markers
+    let distance = distances[i];
+    let marker = markerDistances[distance];
+
+    // Add to the iterator first just in case something fails later to avoid infinite loop
+    i++;
+
+    const marker_lat = marker.position.lat();
+    const marker_lng = marker.position.lng();
+
+    const loc = new google.maps.LatLng(marker_lat, marker_lng);
+    bounds.extend(loc);
+  }
+  map.fitBounds(bounds);       // auto-zoom
+  map.panToBounds(bounds);     // auto-center
```

Comment:
I don't think panToBounds actually does anything after calling fitBounds.

Reply:
This was also copied, can you add this to the list of revisions for later, please?